TestGetObjectPartsAttributes fails

## Current Behavior
```
--- FAIL: TestGetObjectPartsAttributes (0.02s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x70 pc=0xc9fd56]

goroutine 95 [running]:
testing.tRunner.func1.2({0xe2b820, 0x198cf50})
        /usr/lib64/go/1.23/src/testing/testing.go:1632 +0x230
testing.tRunner.func1()
        /usr/lib64/go/1.23/src/testing/testing.go:1635 +0x35e
panic({0xe2b820?, 0x198cf50?})
        /usr/lib64/go/1.23/src/runtime/panic.go:785 +0x132
github.com/nspcc-dev/neofs-s3-gw/api/layer.(*TestNeoFS).SearchObjects(0xe4c300?, {0x126f2f0?, 0xc0004aa870?}, {{0xc0005d80e0}, {0x83, 0x33, 0x61, 0xca, 0xf6, 0x92, ...}, ...})
        <autogenerated>:1 +0x36
github.com/nspcc-dev/neofs-s3-gw/api/layer.(*layer).getFirstArbitraryPart(0xc00005a080, {0x126f2f0, 0xc0004aa870}, {0xc000120649?, 0x7effc62363a8?}, 0xc0004ce960)
        /home/rik/dev/neofs-s3-gw/api/layer/multipart_upload.go:1391 +0x326
github.com/nspcc-dev/neofs-s3-gw/api/layer.(*layer).CompleteMultipartUpload(0xc00005a080, {0x126f2f0, 0xc0004aa870}, 0xc0000a8560)
        /home/rik/dev/neofs-s3-gw/api/layer/multipart_upload.go:678 +0x217
github.com/nspcc-dev/neofs-s3-gw/api/handler.(*handler).CompleteMultipartUploadHandler(0xc0000cc270, {0x126d880, 0xc0004d17c0}, 0xc0000d0c80)
        /home/rik/dev/neofs-s3-gw/api/handler/multipart_upload.go:422 +0x72a
github.com/nspcc-dev/neofs-s3-gw/api/handler.completeMultipartUpload(0xc0000360a0, {0xf5a0ba, 0x15}, {0xf53c18, 0x10}, {0xc0005de240, 0x2c}, {0xc000019f50, 0x1, 0x1})
        /home/rik/dev/neofs-s3-gw/api/handler/encryption_test.go:207 +0x30b
github.com/nspcc-dev/neofs-s3-gw/api/handler.TestGetObjectPartsAttributes(0xc00017cd00)
        /home/rik/dev/neofs-s3-gw/api/handler/attributes_test.go:26 +0x20f
testing.tRunner(0xc00017cd00, 0x1065fd8)
        /usr/lib64/go/1.23/src/testing/testing.go:1690 +0xf4
created by testing.(*T).Run in goroutine 1
        /usr/lib64/go/1.23/src/testing/testing.go:1743 +0x390
FAIL    github.com/nspcc-dev/neofs-s3-gw/api/handler    0.036s
```

## Expected Behavior
Green.

## Possible Solution
Fix the damn thing.

## Context
Tests must be green.

## Regression
Likely.

## Your Environment
* Version of the product used: e0a3f7f6cb8e80069f02c6c58850f8fafa26ec13

--- a/api/layer/neofs_mock.go
+++ b/api/layer/neofs_mock.go
@@ -454,3 +454,30 @@ func getOwner(ctx context.Context) user.ID {
 
 	return user.ID{}
 }
+
+// SearchObjects searches objects with corresponding filters.
+func (t *TestNeoFS) SearchObjects(_ context.Context, prm PrmObjectSearch) ([]oid.ID, error) {
+	var oids []oid.ID
+
+	for _, obj := range t.objects {
+		for _, attr := range obj.Attributes() {
+			for _, f := range prm.Filters {
+				if attr.Key() == f.Header() {
+					switch f.Operation() {
+					case object.MatchStringEqual:
+						if attr.Value() == f.Value() {
+							oids = append(oids, obj.GetID())
+						}
+					case object.MatchStringNotEqual:
+						if attr.Value() != f.Value() {
+							oids = append(oids, obj.GetID())
+						}
+					default:
+					}
+				}
+			}
+		}
+	}
+
+	return oids, nil
+}

